mender-update/daemon/states.cpp

```diff
@@ -363,29 +414,62 @@ void UpdateFailureState::OnEnterUpdateState(Context &ctx, sm::EventPoster<StateE
 			ctx.event_loop, DefaultStateHandler {poster}));
 }
 
-void UpdateSaveArtifactDataState::OnEnterUpdateState(
-	Context &ctx, sm::EventPoster<StateEvent> &poster) {
+static string AddInconsistentSuffix(const string &str) {
+	const auto &suffix = main_context::MenderContext::broken_artifact_name_suffix;
+	// `string::ends_with` is C++20... grumble
```

Comment:
Q: Why not add an `ends_with` and have it default to `std` if `ends_with` is present, like we have done with `unexpected` etc?

Reply:
Unfortunately it's a class member, so we can't add it the way we have done for standalone classes. But I can add it to `common` instead.